authentication/auth_method_client_cert.go

```diff
@@ -68,6 +76,36 @@ func (a servicePrincipalClientCertificateAuth) getAuthorizationToken(sender auto
 	return auth, nil
 }
 
+func (a servicePrincipalClientCertificateAuth) getMSALToken(ctx context.Context, _ autorest.Sender, _ *OAuthConfig, endpoint string) (autorest.Authorizer, error) {
+	certificate, rsaPrivateKey, err := decodePkcs12File(a.clientCertPath, a.clientCertPassword)
+	if err != nil {
+		return nil, fmt.Errorf("decoding pkcs12 certificate: %v", err)
+	}
+
+	environment, err := environments.EnvironmentFromString(a.environment)
+	if err != nil {
+		return nil, fmt.Errorf("environment config error: %v", err)
+	}
+
+	conf := auth.ClientCredentialsConfig{
+		Environment:        environment,
+		TenantID:           a.tenantId,
+		AuxiliaryTenantIDs: a.auxiliaryTenantIds,
+		ClientID:           a.clientId,
+		PrivateKey:         x509.MarshalPKCS1PrivateKey(rsaPrivateKey),
+		Certificate:        certificate.Raw,
+		Scopes:             []string{fmt.Sprintf("%s/.default", strings.TrimRight(endpoint, "/"))},
```

Comment:
not a blocker, more a thought - we should add a helper for `DefaultScopeForEndpoint` which does this